### PR TITLE
Extracted tap config to consistent volume

### PR DESCRIPTION
**Pull request by RoyUP9**

### CODE REVIEW

File: agent/pkg/providers/tap_config_provider.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package providers
```
Comment (IgorGov) [inline agent/pkg/providers/tap_config_provider.go:1]: tapConfig

File: agent/pkg/controllers/config_controller.go [hunk 2]
```diff
@@ -60,7 +59,7 @@ func PostTapConfig(c *gin.Context) {
 	}
 
 	cancelTapperSyncer = cancel
-	globalTapConfig = tapConfig
+	providers.SaveTapConfig(tapConfig)
```
Comment (IgorGov) [inline agent/pkg/controllers/config_controller.go:62]: tapConfig.Save(tapConfig)

File: agent/pkg/providers/tap_config_provider.go (hunk 1 above)
Comment (RoyUP9) [reply agent/pkg/providers/tap_config_provider.go:1]: do we want a folder just for this file?

File: agent/pkg/providers/tap_config_provider.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+func GetTapConfig() *models.TapConfig {
+	if tapConfig == nil {
+		configLock.Lock()
+		defer configLock.Unlock()
+
+		if tapConfig == nil {
+			filePath := fmt.Sprintf("%s%s", shared.DataDirPath, TapConfigFileName)
+
+			content, err := ioutil.ReadFile(filePath)
```
Comment (IgorGov) [inline agent/pkg/providers/tap_config_provider.go:28]: inline "if"

File: agent/pkg/providers/tap_config_provider.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	return tapConfig
+}
+
+func SaveTapConfig(tapConfigToSave *models.TapConfig) {
+	configLock.Lock()
+	defer configLock.Unlock()
+
+	tapConfig = tapConfigToSave
+	data, err := json.Marshal(tapConfig)
+	if err != nil {
```
Comment (IgorGov) [inline agent/pkg/providers/tap_config_provider.go:52]: inline

File: agent/pkg/providers/tap_config_provider.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		logger.Log.Errorf("Error while marshal tap config, err: %v", err)
+	} else {
+		filePath := fmt.Sprintf("%s%s", shared.DataDirPath, TapConfigFileName)
```
Comment (IgorGov) [inline agent/pkg/providers/tap_config_provider.go:55]: lets extract it to const

**Review by IgorGov — APPROVED** [review]

**Pull request merged**